kyaml/setters2/searchreplace_test.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package setters2
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRegexMatch(t *testing.T) {
```

Comment:
So what I'm interested in is seeing how this works with greedy vs lazy quantifiers and capture groups. But this might not be the right place to test that.
But lets make the examples in the tests proper regex expressions. The second example works, but the way the `*` is used is confusing.